File: IWA/Tasks/IWA_5/fib/src/app/fib-form/fib.ts
export class Fib {
    index: bigint;
    value: bigint;

    constructor(index: bigint) {
        this.index = index;
        this.value = Fib.calcFib(index);
    }

    static calcFib(n: bigint): bigint {
        if (n <= 1) {
            const res = BigInt(n);
            return res;
        }
        let fib: bigint = 1n;
        let prevFib: bigint = 1n;

        for (let i: bigint = 2n; i < n; i += 1n) {
            let temp: bigint = fib;
            fib += prevFib;
            prevFib = temp;
        }
        return fib;
    }
}

File: IWA/Tasks/IWA_2/SpringBootJSP_IWA2023/src/main/java/pl/dmcs/springbootjsp_iwa/model/Fib.java
package pl.dmcs.springbootjsp_iwa.model;

public class Fib {

    private String indexString;
    public String getIndexString() {
        return indexString;
    }
    public void setIndexString(String indexString) {
        this.indexString = indexString;
    }

    public static long calcFib(int n) {
        if (n <= 1) {
            return n;
        }
        long fib = 1;
        long prevFib = 1;

        for (int i = 2; i < n; i++) {
            long temp = fib;
            fib += prevFib;
            prevFib = temp;
        }
        return fib;
    }

}
